docs(node-resolve): improve doc related to mainFields (#138)
rollup/plugins: packages/node-resolve/README.md

### `mainFields`

Type: `Array[...String]`<br>
Default: `['module', 'main']`<br>

> Possible values: Array of any values from `['browser', 'jsnext', 'module', 'main']` in any order.

Reviewer: ```suggestion
Valid values: `['browser', 'jsnext', 'module', 'main']`
```

> The fields to scan in a package.json to determine the entry point. Order of fields is significant, first found field would become the resolved entry point. If this list contains "browser", overrides specified in `pkg.browser` will be used.

Reviewer: ```suggestion
Specifies the properties to scan within a `package.json`, used to determine the bundle entry point. The order of property names is significant, as the first-found property is used as the resolved entry point. If the array contains `'browser'`, key/values specified in the `package.json` `browser` property will be used.
```

Author: I didn't backtick package.json, because it wasn't backticked previously. I use backticks to specify field names, like `pkg.browser` (such pkg names was already used before changes).

Reviewer: Hm I think it deserves to be highlighted in that way. I'd like to see this change before I can approve.

### `module`

Type: `Boolean`<br>
Default: `true`

DEPRECATED: use "mainFields" instead

> Use `pkg.module` field for ES6 module if possible. This option takes precedence over both "jsnext" and "main" in the list if such are present.

Reviewer: ```suggestion
```

Let's remove the description of this option, it's Type and Default descriptor as well, but leave the DEPRECATED notice, since it's deprecated.

Author: f4da5de

### `jsnext`

Type: `Boolean`<br>
Default: `false`

DEPRECATED: use "mainFields" instead

> Use `pkg['jsnext:main']` if possible, legacy field pointing to ES6 module in third-party libraries, deprecated in favor of `pkg.module`, see: https://github.com/rollup/rollup/wiki/pkg.module. This option takes precedence over "main" in the list if such is present.

Reviewer: ```suggestion
```

Let's remove the description of this option, it's Type and Default descriptor as well, but leave the DEPRECATED notice, since it's deprecated.

Author: f4da5de

### `main`

Type: `Boolean`<br>
Default: `true`

DEPRECATED: use "mainFields" instead

> Use `pkg.main` field or index.js, even if it's not an ES6 module (needs to be converted from CommonJS to ES6), see https://github.com/rollup/rollup-plugin-commonjs.

Reviewer: ```suggestion
```

Let's remove the description of this option, it's Type and Default descriptor as well, but leave the DEPRECATED notice, since it's deprecated.

Author: f4da5de

### `browser`

Type: `Boolean`<br>
Default: `false`

> Some package.json files have a `pkg.browser` field which specifies alternative files to load for people bundling for the browser. If that's you, either use this option or add "browser" to the "mainFields" option, otherwise `pkg.browser` will be ignored. This option takes precedence over any other in "mainFields" if such are present.

Reviewer: ```suggestion
If `true`, instructs the plugin to use the `"browser"` property in `package.json` files to specify alternative files to load for bundling. This is useful when bundling for a browser environment. Alternatively, a value of `'browser'` can be added to the `mainFields` option. If `false`, any `"browser"` properties in package files will be ignored. This option takes precedence over `mainFields`.
```

This rewording matches the format for other Boolean options in the plugins repo.
